### PR TITLE
use the standard resource sync to reduce code complexity

### PR DESCRIPTION
**Pull request by deads2k**
spelling changes, but the output is the same

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -5,13 +5,15 @@ import (
 	"os"
 	"time"
 
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/operatorclient"
```
Comment (mfojtik) [inline pkg/operator/starter.go]: nit: these should go down

File: pkg/operator/resourcesynccontroller/resourcesynccontroller.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/client-go/kubernetes"
+)
+
+func NewResourceSyncController(
```
Comment (mfojtik) [inline pkg/operator/resourcesynccontroller/resourcesynccontroller.go]: nit: don't hate godoc

File: pkg/operator/workload_controller.go [hunk 3]
```diff
@@ -4,6 +4,8 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/operatorclient"
```
Comment (mfojtik) [inline pkg/operator/workload_controller.go]: nit: order

File: pkg/operator/workload_controller.go [hunk 4]
```diff
@@ -204,7 +206,7 @@ func (c *OpenShiftAPIServerOperator) namespaceEventHandler() cache.ResourceEvent
 			if !ok {
 				c.queue.Add(workQueueKey)
 			}
-			if ns.Name == targetNamespaceName {
+			if ns.Name == operatorclient.TargetNamespaceName {
```
Comment (mfojtik) [inline pkg/operator/workload_controller.go]: nit: `operatorclient.IsTargetNamespace(ns.Name)`

File: pkg/operator/workload_controller_openshiftapiserver_v311_00.go [hunk 5]
```diff
@@ -61,24 +63,13 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
...
-	}
-
 	imageImportCAModified, err := manageOpenShiftAPIServerImageImportCA_v311_00_to_latest(c.openshiftConfigClient, c.kubeClient.CoreV1(), c.eventRecorder)
 	if err != nil {
 		errors = append(errors, fmt.Errorf("%q: %v", "client-ca", err))
 	}
 
 	forceRollingUpdate = forceRollingUpdate || operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration
-	forceRollingUpdate = forceRollingUpdate || configMapModified || etcdModified || clientCAModified || imageImportCAModified
+	forceRollingUpdate = forceRollingUpdate || configMapModified || imageImportCAModified
```
Comment (mfojtik) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00.go]: glad to see this getting shorter

File: pkg/operator/workload_controller_openshiftapiserver_v311_00_test.go [hunk 6]
```diff
@@ -5,6 +5,8 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/operatorclient"
```
Comment (mfojtik) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00_test.go]: nit: order

File: pkg/operator/operatorclient/interfaces.go [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package operatorclient
+
+const (
+	EtcdNamespaceName                     = "kube-system"
+	KubeAPIServerNamespaceName            = "openshift-kube-apiserver"
+	UserSpecifiedGlobalConfigNamespace    = "openshift-config"
```
Comment (sttts) [inline pkg/operator/operatorclient/interfaces.go]: GlobalUserSpecifiedConfigNamespace. The config is user specified, not the namespace.

**Comment by mfojtik** [comment]
@deads2k just nits, i'm fine fixing them in next iteration.

File: pkg/operator/operatorclient/interfaces.go [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package operatorclient
+
+const (
+	EtcdNamespaceName                     = "kube-system"
+	KubeAPIServerNamespaceName            = "openshift-kube-apiserver"
```
Comment (sttts) [inline pkg/operator/operatorclient/interfaces.go]: OperandNamespace

File: pkg/operator/operatorclient/informers.go [hunk 9]
```diff
@@ -0,0 +1,28 @@
+package operatorclient
```
Comment (sttts) [inline pkg/operator/operatorclient/informers.go]: why not just `client`?

File: pkg/operator/resourcesynccontroller/resourcesynccontroller.go [hunk 10]
```diff
@@ -34,28 +34,8 @@ func NewResourceSyncController(
...
 	}
 	if err := resourceSyncController.SyncConfigMap(
-		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.TargetNamespaceName, Name: "aggregator-client-ca"},
-		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.MachineSpecifiedGlobalConfigNamespace, Name: "aggregator-client-ca"},
-	); err != nil {
-		return nil, err
-	}
-	if err := resourceSyncController.SyncConfigMap(
-		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.OperatorNamespace, Name: "initial-client-ca"},
-		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.UserSpecifiedGlobalConfigNamespace, Name: "initial-client-ca"},
```
Comment (sttts) [inline pkg/operator/resourcesynccontroller/resourcesynccontroller.go]: where is that gone?

**Comment by danwinship** [comment]
> Looks like networking
> 
> ```
> piservice/v1.apps.openshift.io: not available: no response from https://172.30.200.221:443: Get https://172.30.200.221:443: dial tcp 172.30.200.221:443: connect: no route to host
> ```

"no route to host" on a service IP usually means "the service has no functioning endpoints".

(I don't see that error in any of the test run logs linked to this PR though?)

**Comment by deads2k** [comment]
> "no route to host" on a service IP usually means "the service has no functioning endpoints".
> 
> (I don't see that error in any of the test run logs linked to this PR though?)

events and pods indicate a healthy pod and container.

File: pkg/operator/operatorclient/interfaces.go (hunk 7 above)
Comment (deads2k) [reply pkg/operator/operatorclient/interfaces.go]: > GlobalUserSpecifiedConfigNamespace. The config is user specified, not the namespace.

will do either later or if I don't go green.

File: pkg/operator/resourcesynccontroller/resourcesynccontroller.go (hunk 10 above)
Comment (deads2k) [reply pkg/operator/resourcesynccontroller/resourcesynccontroller.go]: bad copy paste that I guess I didn't need later :(

**Comment by deads2k** [comment]
> "no route to host" on a service IP usually means "the service has no functioning endpoints".
> 
> (I don't see that error in any of the test run logs linked to this PR though?)

Also, we indicate lack of endpoints as 
```
not available: endpoints for service/api 
```

**Comment by danwinship** [comment]
So where is this error occurring? As I said, I didn't see that message in any of the test run logs.

**Comment by deads2k** [comment]
> So where is this error occurring? As I said, I didn't see that message in any of the test run logs.

You can see the output in our clusteroperator here: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-openshift-apiserver-operator/115/pull-ci-openshift-cluster-openshift-apiserver-operator-master-e2e-aws/512/artifacts/e2e-aws/clusteroperators.json

It is the result from the apiservice which is produced when the kube-apiserver tries to contact the openshift-apiserver service.

**Comment by deads2k** [comment]
merged in a different pull, but the service network problem is still concerning.

**Pull request closed without merging**